### DISCUSSION TITLE
Recurring due dates

### DISCUSSION THREAD

**Discussion by AhmedA1559** [jotaen/xit#5] [File Format]
Item could have recurring due dates, good for representing for situations such as daily tasks.

The specification for [repeating intervals](https://en.wikipedia.org/wiki/ISO_8601#Repeating_intervals) could be implemented from ISO 8601 like so:
`
[ ] Flower the plants -> R/2022-04-01T21:00:00Z/P1Y1M0DT0H0M
`

**Comment by jotaen** [comment]
Interesting use case. Would the item status (`[ ]`) stay open then for the entire time? Or how could one keep track of whether the item has been taken care of or not?

**Reply by global-node** [reply]
what about [@] for ongoing task or something like [r] for recurring?

**Reply by rstephens1** [reply]
I was thinking a similar thing. ISO8601 and perhaps [...] for recurring. As for completed, this is where things get complicated. I am building [ChainList](https://www.chainlist.com/) - and we use the concept of "template" for the source list, and when you "do" or "run" an instance of a list (or recurring instance, we refer to that as a "checklist". We used to use the term "checklist" and "Run", but Template as source and "checklist" as "run" seem to be more intuitive for users.

(I'm watching this because as we look at export options, we already were looking at ISO8601 and an .xit option would be nice to offer as well)

**Reply by jotaen** [reply]
I think recurrence is an example where a plain text file format falls short, so I’m not sure there is a good way for it to be added to the spec. One potential approach to go about this – albeit a bit unsatisfactory maybe – would be to solve recurrence through tooling, e.g. to have a tool automatically add (templated) items or item groups once per week/month/whatever-interval to the file.
